File: 0278-first-bad-version/0278-first-bad-version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
       if(n == 1){
        return 1;
       } 
       int l =0;
       int h = n-1;
       while(l <= h){
        int mid = l+(h-l)/2;
        if(isBadVersion(mid)){
            h = mid -1;
        }
        else{
            l = mid + 1;
        }
       }
       return l;
    }
};

File: 0278-first-bad-version/0278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int l = 0;
        int h = n;
        int mid ;
        
        while(l <= h){
            mid = l  + (h - l)/2;
            if(isBadVersion(mid) == false){
                l = mid + 1;
            }
            else{
                h = mid - 1;
            }
        }
        return l;
    }
}